vac/raw/eth-mls-offchain.md

```diff
@@ -204,32 +231,162 @@ including welcome messages for the new members. Therefore, the `commit message`
+and detect any unauthorized steward commits.
+3. `Emergency criteria proposal`: If there is a malicious member or steward,
+this event MUST be voted on to finalize it.
+If this returns YES, the next epoch MUST include the removal of the member or steward.
+Proposal.payload MUST consist of the evidence of the dishonesty as described in the Steward violation list,
+and the identifier of the malicious member or steward.
+This proposal can be created by any member in any epoch.
+
+The order of consensus proposal messages is important to achieving a consistent result.
+Therefore, messages MUST be prioritized by type in the following order, from highest to lowest priority:
+
+- `Emergency criteria proposal`
+
+- `Steward election proposal`
 
-### Multi steward with single consensus
+- `Commit proposal`
+
+This means that if a higher-priority consensus proposal is present in the network,
+lower-priority messages MUST be withheld from transmission until the higher-priority proposals have been finalized.
+
+### Steward list creation
+
+The `steward list` size as `sn` is determined when the group is created.
+The index of the slots shows epoch info and value of index shows `member id`s.
+The next in line steward for the `epoch E` is named as `epoch steward`, which has index E.
+And the subsequent steward in the `epoch E` is named as the `backup steward`.
+For example, let's assume steward list is (S3, S2, S1) if in the previous epoch the roles were
+(`backup steward`: S2, `epoch steward`: S1), then in the next epoch they become
+(`backup steward`: S3, `epoch steward`: S2) by shifting.
+
+If the `epoch steward` is honest, the `backup steward` does not involve the process in epoch,
+and the `backup steward` will be the `epoch steward` within the `epoch E+1`.
+
+If the `epoch steward` is malicious, the `backup steward` is involved in the commitment phase in `epoch E`
+and the former steward becomes the `backup steward` in `epoch E`.
+
+Liveness criteria:
+
+Once the active `steward list` has completed its assigned epoch,
+members MUST proceed to elect the next set of stewards
+(which MAY include some or all of the previous members).
+This election is conducted through a type 2 consensus procedure, `steward election proposal`.
+
+A `Steward election proposal` is considered valid only if the resulting `steward list`
+is produced through a deterministic process that ensures an unbiased distribution of steward assignments,
+since allowing bias could enable a malicious participant to manipulate the list
+and retain control within a favored group for multiple epochs.
+
+The list MUST be composed of the first `sn` members from the member list,
+sorted according to the ascending value of `SHA256(epoch E || member id)`,
+where `epoch E` is the epoch in which the election proposal is initiated.
+Any proposal with a list that does not adhere to this generation method MUST be rejected by all members.

```

Comment:
I love how this is deterministic as it means every client can derive the same steward list. However I'm curious about the case where the current steward list contains only "Stale" members. (Members who are are valid members of the group, but remain offline for an extended period of time; Either due to unavailability, or a fault causing a desynchronization in state)
- The larger the group the more likely that stale clients exist
- During this time proposals build up, but there is no steward available to generate valid commits.
- Lack of commits results in a partial deadlock, until the era expires and a new steward election proposal occurs and an "online" steward is selected.

Is the intention that applications must prune inactive members to avoid this case?